### ISSUE TITLE
[FEAT.] : Upgrade version

### ISSUE THREAD

**Issue by Sibyrd** [Zagrios/bs-manager#573]
## Problem
Feature request - please add an option to upgrade your BS version without having to install another instance so mods dont have to be reinstalled.

## Additional context
If this already exists I apologise I honestly cannot find it at all

**Comment by zylantha** [comment]
+1 for this feature.  I would really like to:

-  have an option to show which of the mods installed in a version are compatible / incompatible with another alternate (newer) version.
- re-install all of the mods installed in one version into the new version after installation.

Currently it is really very tricky to be able to tell which mods will be unavailable in a new BS version, and it's really difficult to tell which is the latest version that will work with all of my mods.

Ideally I could say "show me my mod compatibility with other beat saber versions", I can then either select specific versions or have it show all newer versions, and then it shows a list of all of the mods and whether they are available or not.  The unavailability of some mods may not matter to me, but others (like Chroma, Noodles, Cinema) do - and while the data for compatibility against each version exists, there's no easy way for me to digest this. 

It would be a killer feature to have!

**Comment by silentrald** [comment]
@zylantha

Yeah, was kinda looking into #281 and #285 which could be a part of this feature. This is sort of similar to https://slinkstr.com/beat-saber/beatmods-comparison/. I haven't thought of a UI/UX of this but I'll try to start once I get approval from the owner ig.

For the custom external mods, already got a pre-requisite PR #569 for this. Only implemented right now is the drag n drop feature, although I'm planning to add custom mod tracking so that its easier to see and manage those mod files. Still waiting for it to get merged before I can start coding this.

**Comment by dirtball-official** [comment]
What if in each installed version's Mods screen there is a drop-down menu listing the other installed versions and a "Select Mods" (or whatever) button? When you click that button, it will check all the boxes in the current version that are checked in the version you selected from the drop-down menu, at least all the ones that are available. Then you can review / edit the list and click "Install or update" like usual. This would make upgrades much less painful and should hopefully avoid (most) dependency issues.
